src/blk/zoned/HMSMRDevice.cc

```diff
@@ -412,6 +412,14 @@ void HMSMRDevice::_detect_vdo()
   return;
 }
 
+void HMSMRDevice::reset_zones(const std::set<uint64_t> *zones) {
```

Comment:
nit, if `zones` is never a nullptr, why not passing a const reference?

Reply:
Done.